File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        
        string = "".join(filter(str.isalnum, s.lower()))
        
        begin, end = 0, len(string) - 1
        
        if len(string) <= 1:
            return True
        
        while begin < end:
            if string[begin] != string[end]:
                return False
            else:
                begin += 1
                end -= 1
                
        return True

File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        
     s.erase(std::remove_if(s.begin(), s.end(), [](char c) { return !std::isalnum(c); }), s.end());
    std::transform(s.begin(), s.end(), s.begin(), ::tolower);

    int begin = 0;
    int end = s.length() - 1;

    if (s.length() <= 1) {
        return true;
    }

    while (begin < end) {
        if (s[begin] != s[end]) {
            return false;
        } else {
            begin++;
            end--;
        }
    }

    return true;
    }
};